flow-build-info.json isn't found in production build uberjar

Related to https://github.com/vaadin/flow/issues/6965

The production uberjar fails with

```
Caused by: java.lang.IllegalStateException: Unable to determine mode of operation. To use npm mode, ensure 'flow-build-info.json' exists on the classpath. With Maven, this is handled by the 'prepare-frontend' goal. To use compatibility mode, add the 'flow-server-compatibility-mode' dependency. If using Vaadin with Spring Boot, instead set the property 'vaadin.compatibilityMode' to 'true' in 'application.properties'.
	at com.vaadin.flow.server.DeploymentConfigurationFactory.verifyMode(DeploymentConfigurationFactory.java:382)
	at com.vaadin.flow.server.DeploymentConfigurationFactory.lambda$readBuildInfo$1ac205d3$1(DeploymentConfigurationFactory.java:230)
	at com.vaadin.flow.server.DefaultDeploymentConfiguration.checkCompatibilityMode(DefaultDeploymentConfiguration.java:275)
	at com.vaadin.flow.server.DefaultDeploymentConfiguration.<init>(DefaultDeploymentConfiguration.java:120)
```

I can reproduce this issue easily with my https://github.com/mvysny/vaadin14-embedded-jetty project:

* By default it uses 14.0.12 and it works: build the project with `mvn -C clean package -Pproduction` then go to `target/` and run `java -jar vaadin14-embedded-jetty-1.0-SNAPSHOT-uberjar.jar` and browse to localhost:8080 -  everything works.
* However, edit the pom.xml and set vaadin version to 14.0.13 or 14.0.14, build the project with `mvn -C clean package -Pproduction` then go to `target/` and run `java -jar vaadin14-embedded-jetty-1.0-SNAPSHOT-uberjar.jar` and browse to localhost:8080 - it will crash.

The `META-INF/VAADIN/config/flow-build-info.json` exists in both jars, however it seems it's being ignored somehow in 14.0.13 and 14.0.14...

Sounds like 14.0.13+ only checks `fragment.xml` (which is for some reason ignored by Jetty in uberjar mode).

--- a/flow-server/src/main/java/com/vaadin/flow/server/DeploymentConfigurationFactory.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/DeploymentConfigurationFactory.java
@@ -31,6 +31,7 @@ import java.util.Optional;
 import java.util.Properties;
 
 import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.vaadin.flow.component.UI;
@@ -87,6 +88,8 @@ public final class DeploymentConfigurationFactory implements Serializable {
 
     public static final String DEV_FOLDER_MISSING_MESSAGE = "Running project in development mode with no access to folder '%s'.%n"
             + "Build project in production mode instead, see https://vaadin.com/docs/v14/flow/production/tutorial-production-mode-basic.html";
+    private static final Logger logger = LoggerFactory
+            .getLogger(DeploymentConfigurationFactory.class);
 
     private DeploymentConfigurationFactory() {
     }
@@ -330,11 +333,10 @@ public final class DeploymentConfigurationFactory implements Serializable {
         if (Boolean.toString(parsedBoolean).equalsIgnoreCase(booleanString)) {
             return parsedBoolean;
         }
-        LoggerFactory.getLogger(DeploymentConfigurationFactory.class)
-                .debug(String
-                        .format("Property named '%s' is boolean, but contains incorrect value '%s' that is not boolean '%s'",
-                                SERVLET_PARAMETER_PRODUCTION_MODE,
-                                booleanString, parsedBoolean));
+        logger.debug(String.format(
+                "Property named '%s' is boolean, but contains incorrect value '%s' that is not boolean '%s'",
+                SERVLET_PARAMETER_PRODUCTION_MODE, booleanString,
+                parsedBoolean));
         return false;
     }
 
@@ -363,6 +365,23 @@ public final class DeploymentConfigurationFactory implements Serializable {
                     return FrontendUtils.streamToString(resource.openStream());
                 }
             }
+        } else {
+            URL firstResource = resources.get(0);
+            if (resources.size() > 1) {
+                String warningMessage = String
+                        .format("Unable to fully determine correct flow-build-info.%n"
+                                        + "Accepting file '%s' first match of '%s' possible.%n"
+                                        + "Please verify flow-build-info file content.",
+                                firstResource.getPath(), resources.size());
+                logger.warn(warningMessage);
+            } else {
+                String debugMessage = String
+                        .format("Unable to fully determine correct flow-build-info.%n"
+                                        + "Accepting file '%s'",
+                                firstResource.getPath());
+                logger.debug(debugMessage);
+            }
+            return FrontendUtils.streamToString(firstResource.openStream());
         }
         // No applicable resources found.
         return null;
@@ -422,7 +441,7 @@ public final class DeploymentConfigurationFactory implements Serializable {
         if (!hasTokenFile && hasWebpackConfig) {
             // the current working directory will be used automatically by the
             // dev server unless it's specified explicitly
-            LoggerFactory.getLogger(DeploymentConfigurationFactory.class).warn(
+            logger.warn(
                     "Found 'webpack.config.js' in the project/working directory. "
                             + "Will use it for webpack dev server.");
         }

